python_modules/libraries/dagster-snowflake-pandas/dagster_snowflake_pandas/snowflake_pandas_type_handler.py

```diff
@@ -86,16 +53,23 @@ def handle_output(
         connector.paramstyle = "pyformat"
         with _connect_snowflake(context, table_slice) as con:
             with_uppercase_cols = obj.rename(str.upper, copy=False, axis="columns")
-            with_uppercase_cols = with_uppercase_cols.apply(
-                _convert_timestamp_to_string, axis="index"
-            )
-            with_uppercase_cols.to_sql(
-                table_slice.table,
-                con=con.engine,
-                if_exists="append",
-                index=False,
-                method=pd_writer,
-            )
+
+            try:
+                with_uppercase_cols.to_sql(
+                    table_slice.table,
+                    con=con.engine,
+                    if_exists="append",
+                    index=False,
+                    method=pd_writer,
+                )
+            except InterfaceError as e:
```

Comment:
another option instead of reacting to a failed write would be to pre-emtively check if the dataframe as Timestamp dat. if so we check if there is timezone information, and raise an error if it doesn't have that info